### DISCUSSION TITLE
How to use the assets directory?

### DISCUSSION THREAD

**Discussion by DenuxPlays** [leptos-rs/leptos#2060] [Q&A] [answered]
Hello,

I try to use a picture from the asset directory.
Do I have to "import" it or something like that because `/picture.svg` doesn't work.
The image can be found in `target/site` but I cannot access it.
The favicon can be loaded but my image not.

**Comment by gbj** [comment, marked as answer]
This kind of depends on how your server is set up exactly.

The cargo-leptos docs say:
```
# Assets source dir. All files found here will be copied and synchronized to site-root.
assets-dir = "assets"
```
This means that whatever is in the `assets` directory will be copied to the `site-root` -- in the default case, `target/site`.

Then the server can be set up to server static files from that directory. For example, if I look at the Actix starter template, it serves this `site-root` directory at `/assets`:
```rust

        App::new()
            .route("/api/{tail:.*}", leptos_actix::handle_server_fns())
            // serve JS/WASM/CSS from `pkg`
            .service(Files::new("/pkg", format!("{site_root}/pkg")))
            // serve other assets from the `assets` directory
            .service(Files::new("/assets", site_root))
```
https://github.com/leptos-rs/start/blob/main/src/main.rs#L20-L25

So I can access a static file `foo.txt` in `assets` at `/assets/foo.txt`

**Reply by DenuxPlays** [reply]
Tanks :)
